src/app/customer/CreateAccountForm.tsx

```diff
@@ -103,13 +104,14 @@ export default withLanguage(withFormik<CreateAccountFormProps & WithLanguageProp
     handleSubmit: (values, { props: { onSubmit = noop } }) => {
         onSubmit(values);
     },
-    mapPropsToValues: () => ({
+    mapPropsToValues: ({requiresMarketingConsent}) => ({
         firstName: '',
         lastName: '',
         email: '',
         password: '',
         customFields: {},
         acceptsMarketingEmails: [],
```

Comment:
I think what you're looking for is to provide a different default value for `acceptsMarketingEmails` if `requiresMarketingConsent` is `false`.